contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -90,6 +90,8 @@ contract ValidatorShare is IValidatorShare {
 
     function buyVoucher(uint256 _amount) public onlyWhenUnlocked {
         uint256 share = _amount.mul(100).div(exchangeRate());
+        require(share > 0, "Insufficient amount to buy share ");
```

Comment:
Also, Please add a check `require(delegators[msg.sender].amount==0)` for MAT1142

Reply:
done